### PR TITLE
Fixed the agent installation issue when server service is of Type NodePort/LoadBalancer

### PR DESCRIPTION
**Pull request by Jonsy13**
Signed-off-by: Jonsy13 <vedant.shrotria@chaosnative.com>

## Proposed changes

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

This PR will fix the issue of the `Subscriber not getting deployed when the Backend Server service is of type NodePort`.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [X] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [X] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Review by ishangupta-ds — APPROVED** [review]

**Comment by rajdas98** [comment]
@Jonsy13 is checking for load balancer service type case. Lets not merge this PR now

**Review by gdsoumya — APPROVED** [review]

**Review by rajdas98 — APPROVED** [review]

File: litmus-portal/graphql-server/pkg/k8s/cluster.go [hunk 1]
```diff
@@ -180,9 +180,14 @@ func GetServerEndpoint() (string, error) {
 		switch exp {
 		case "loadbalancer":
 			log.Print("loadbalancer")
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/k8s/cluster.go:182]: Is this log needed anymore?

File: litmus-portal/graphql-server/pkg/k8s/cluster.go [hunk 2]
```diff
@@ -180,9 +180,14 @@ func GetServerEndpoint() (string, error) {
...
-				return "", errors.New("ExternalIP is not present for loadbalancer service type")
+			if len(svc.Status.LoadBalancer.Ingress) > 0 {
+				if svc.Status.LoadBalancer.Ingress[0].Hostname != "" {
+					IPAddress = svc.Status.LoadBalancer.Ingress[0].Hostname
+				} else if svc.Status.LoadBalancer.Ingress[0].IP != "" {
+					IPAddress = svc.Status.LoadBalancer.Ingress[0].IP
+				} else {
+					return "", errors.New("ExternalIP is not present for loadbalancer service type")
+				}
 			}
```
Comment (gdsoumya) [inline litmus-portal/graphql-server/pkg/k8s/cluster.go:191]: in else condition we should return an error because the IPAddress is empty?

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/pkg/k8s/cluster.go [hunk 3]
```diff
@@ -180,9 +180,14 @@ func GetServerEndpoint() (string, error) {
...
-			IPAddress = svc.Spec.LoadBalancerIP
-			if IPAddress == "" {
-				return "", errors.New("ExternalIP is not present for loadbalancer service type")
+			if len(svc.Status.LoadBalancer.Ingress) > 0 {
+				if svc.Status.LoadBalancer.Ingress[0].Hostname != "" {
+					IPAddress = svc.Status.LoadBalancer.Ingress[0].Hostname
+				} else if svc.Status.LoadBalancer.Ingress[0].IP != "" {
+					IPAddress = svc.Status.LoadBalancer.Ingress[0].IP
+				} else {
+					return "", errors.New("ExternalIP is not present for loadbalancer service type")
```
Comment (ishangupta-ds) [inline litmus-portal/graphql-server/pkg/k8s/cluster.go:189]: Load balancer IP or hostname is not... @Jonsy13

File: litmus-portal/graphql-server/pkg/k8s/cluster.go (hunk 2 above)
Comment (Jonsy13) [reply litmus-portal/graphql-server/pkg/k8s/cluster.go:191]: @gdsoumya  yes, it should return an error. Thanks for pointing this out.

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: litmus-portal/graphql-server/pkg/k8s/cluster.go (hunk 1 above)
Comment (Jonsy13) [reply litmus-portal/graphql-server/pkg/k8s/cluster.go:182]: No, i can remove it if not required.

File: litmus-portal/graphql-server/pkg/k8s/cluster.go (hunk 3 above)
Comment (Jonsy13) [reply litmus-portal/graphql-server/pkg/k8s/cluster.go:189]: Sure @ishangupta-ds making this change.

**Review by gdsoumya — APPROVED** [review]

**Pull request merged**